gloo/transport/tcp/error.cc

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <gloo/transport/tcp/error.h>
+
+#include <cstring>
+#include <sstream>
+
+namespace gloo {
+namespace transport {
+namespace tcp {
+
+const Error Error::kSuccess = Error();
+
+std::string Error::what() const {
+  return "no error";
+}
+
+std::string SystemError::what() const {
+  std::stringstream ss;
```

Comment:
nit: maybe std::ostringstream for these?

(since this is error path, it shouldn't actually matter, but std::streamstream effectively has to pay the unused std::istreamstream constructor/destructor cost, which actually was showing up in my serialization hot-path profiles the other month)

Reply:
Good point! Updated to use `std::ostringstream`.